docs/checks.md

```diff
@@ -0,0 +1,299 @@
+13. [function-name](checks.md#naming)
+14. [data-type-name](checks.md#naming)
+15. [column-name](checks.md#naming)
+16. [parameter-name](checks.md#naming)
+17. [constant-name](checks.md#naming)
+18. [join](checks.md#join)
+19. [imports](checks.md#imports)
+
+## parser-failed
+Checks if ZetaSQL parser succeeds to parse your sql statements. 
+
+**Example**
+```sql
+-- Here is an OK statement.
+SELECT *;
+-- Here is an invalid statement.
+SELECT A FROM B C D;
+```
+**Linter Output**
+```
+In line 4, column 19: Syntax error: Expected end of input but got identifier "D" [parser-failed]
+```
+
+## line-limit-exceed
+Each line in an SQL file should contain less characters than a configurable [line limit](config.md#config).
+
+**Example**
+```sql
+-- Here is a very long comment that will exceed line limit and should be seperated into different lines.
+SELECT *;
+```
+**Linter Output**
+```
+In line 1, column 101: Lines should be <= 100 characters long [line-limit-exceed]
+```
+
+## statement-semicolon
+Checks if every SQL statement ends with a semicolon (';')
+
+**Example**
+```sql
+-- Here is a normal statement that ends with a semicolon.
+SELECT *;
+-- Here is also valid statement but it is missing a semicolon.
+SELECT A FROM B
+```
+**Linter Output**
+```
+In line 4, column 16: Each statement should end with a semicolon ';' [statement-semicolon]
+```
+## consistent-letter-case
+Checks if every keyword, either all uppercase or all lowercase.
+
+**Example**
+```sql
+-- The 'SELECT' keyword written as all uppercase, so there is no problem.
+SELECT *;
+
+-- Here, 'FROM' keyword consists of uppercase and lowercase letters,
+-- which will cause a lint error.
+SELECT A From B;
+```
+**Linter Output**
+```
+In line 6, column 10: All keywords should be either uppercase or lowercase [consistent-letter-case]
+```
+## consistent-comment-style
+ZetaSQL linter supports different kind of comment styles ('#', '//', or '--'). Only one of them should be used consistently,
+
+**Example**
+```sql
+-- The first comment used '--'
+SELECT *;
+# The second comment used '#', which is different than the first comment
+# and it will cause lint error.
+```
+
+**Linter Output**
+```
+In line 3, column 1: One line comments should be consistent, expected: --, found: # [consistent-comment-style]
+In line 4, column 1: One line comments should be consistent, expected: --, found: # [consistent-comment-style]
+```
+## alias
+Every alias should use keyword 'AS'. 
+
+**Example**
+```sql
+-- Here is a valid usage.
+SELECT A AS a;
+
+-- Here 'AS' keyword is omitted. Still a valid SQL statement, but linter will complain.
+SELECT A a;
+```
+
+**Linter Output**
+```
+In line 5, column 10: Always use AS keyword before aliases [alias]
+```
+## uniform-indent
+Indentation character should be uniform. Generally, either tabs or spaces are used. It is configurable with variable [allowed_indent](config.md#config). By default, spaces are expected to be used as the indentation character.
```

Comment:
Indentation character should be uniform. It could be either tab or space.

Reply:
Can be, or Could be?